### ISSUE TITLE
Build failure: python3Packages.faiss (on Darwin)

### ISSUE THREAD

**Issue by sarahec** [NixOS/nixpkgs#340842]
### Steps To Reproduce

1. On Darwin...
2. build `python312Packages.faiss`
3. Observe:
```sh
make: *** [Makefile:208: swigfaiss] Error 2
error: builder for '/nix/store/mbr9lq6ppqzws7q2qv6s70fhrhrzjl3z-faiss-1.8.0.drv' failed with exit code 2;
       last 10 log lines:
       >         return SWIG_NewPointerObj(data, SWIGTYPE_p_unsigned_long, 0);
       >                                         ^
       > /tmp/nix-build-faiss-1.8.0.drv-0/source/build/faiss/python/CMakeFiles/swigfaiss.dir/swigfaissPYTHON_wrap.cxx:5095:41: error: use of undeclared identifier 'SWIGTYPE_p_long'
       >         return SWIG_NewPointerObj(data, SWIGTYPE_p_long, 0);
       >                                         ^
       > 2 errors generated.
       > make[3]: *** [faiss/python/CMakeFiles/swigfaiss.dir/build.make:76: faiss/python/CMakeFiles/swigfaiss.dir/CMakeFiles/swigfaiss.dir/swigfaissPYTHON_wrap.cxx.o] Error 1
       > make[2]: *** [CMakeFiles/Makefile2:269: faiss/python/CMakeFiles/swigfaiss.dir/all] Error 2
       > make[1]: *** [CMakeFiles/Makefile2:276: faiss/python/CMakeFiles/swigfaiss.dir/rule] Error 2
       > make: *** [Makefile:208: swigfaiss] Error 2
       For full logs, run 'nix log /nix/store/mbr9lq6ppqzws7q2qv6s70fhrhrzjl3z-faiss-1.8.0.drv'.
error: 1 dependencies of derivation '/nix/store/wnjfqxjfakg3m3ymdf9bf661zm3rzxq2-python3.12-faiss-1.8.0.drv' failed to build
```

This is a known issue when swig generates Darwin code: https://github.com/swig/swig/issues/2864
Faiss accepted a patch for the next release: https://github.com/facebookresearch/faiss/commit/bd22c936e108d4551648435764b126aef40b4530

### Notify maintainers

@SomeoneSerge 
cc: @natsukium 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.6.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by sarahec** [comment]
Unfortunately, cherry-picking the patch didn't work. I'll close this for now as I don't see a way to patch around it; I'll see if I can lobby for an interim faiss release.

**Comment by SomeoneSerge** [comment]
So the issue is still present? Then we should keep this open?

**Comment by sarahec** [comment]
Yes, it's still present so let's keep this open.

I've asked the faiss team if they could cut a release to include Darwin. We'll see what they say.

Otherwise, I'll keep looking into potential patches.

**Comment by sarahec** [comment]
See https://github.com/facebookresearch/faiss/issues/3846

**Comment by sarahec** [comment]
@SomeoneSerge they are producing a new release next week!

**Comment by sarahec** [comment]
Closed by #341418

**Comment by sarahec** [comment]
Closed by #341418
